### DISCUSSION TITLE
GitHub Actions (and template) verifier

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#836] [Ideas]
It would be nice, if https://metrics.lecoq.io/ could offer a sort of verifier for GitHub Actions workflows and/or Markdown files (in case a `markdown` template is used).

What this verifier would do is essentially simulating a GitHub Actions run, telling you what it would've generated and what other steps it would have done.
In the case of also providing a markdown template would it show you the output with some dummy data (to not overuse requests) and inform you about any errors such as wrong configuration options.

***Why not use actions with json output or smth?***
Because that not only has the risk of failed runs, which are annoying on their own, but also requires you to actually do such runs, which in case of adding stuff through PRs really won't help here.

Please treat this as a low priority thing, because the last thing I want is having fixes and new changes to the action being postponed by this here. :)

**Comment by lowlighter** [comment]
Hi 👋 !

It would be a nice feature to have definitely, though it'd probably require a fair amount of work to implement.
Plus there are some stuff that cannot be tested with the web action, and although it's able to render repo/markdown templates if you know how to do it, it's kind of "hidden" since the web app currently doesn't have the ui to do so.

I personnally don't think that I'll have time and courage to work on it, but if someone wants to give it a shot they can open a pull request for it.

However, since the recent introduction of [`config_presets`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#-using-presets) option, I'd like to promote this feature by providing "ready-to-go" configurations with pre-configured templates and options (especially for repositories), rather than make users overhelmed by all available options

**Reply by Andre601** [reply]
Understandable.

One thing I wonder tho... Is the contributors plugin not listed on the website, or is it under a diffeent name?

**Reply by lowlighter** [reply]
The website does not display any plugin which has only `supports: ["repository"]` in their `metadata.yml` (so all plugins which don't support either users or orgs accounts)

It's related to what I was saying about some stuff that cannot be tested with the web action. The `repository` support was added later but the web ui was never updated to support it 😕

**Comment by Andre601** [comment]
Something I was thinking about just now:
There is the output option for JSON, right?
Maybe have an option such as `log` which would essentially work as a dry-run where the action acts as if it was working normal (while not actually pushing stuff), to log everything into a TXT file to the examine by the maintainer.

The TXT would give detailed insights as to what the action did and where it would most likely fail and why.

I'm aware that Actions do have logs, but I feel like they are kind of tedious to work with in the long run...

**Reply by lowlighter** [reply]
You can do something similar using [`dryrun`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#dryrun) and [`debug`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#debug) options, and use the "raw logs" of the action instead (clicking on the gear icon) to avoid the "tediousity" of using the web interface of actions to nagivate through logs.

It should more or less cover your use case I guess
